versioned_docs/version-5.x/basics/upgradeability.md

```diff
@@ -271,6 +271,29 @@ As you can see, delegatee's code looks like a normal ink! Smart Contract with so
 - `addresses` mapping key is identical
 - Constructor does not have any logic, as the code is never instantiated. (It can be, but plays no effect on the execution)
 
+### Delegate dependency locks
+
+The `delegator` contract depends upon the contract code to which it delegates. Since code
+can be deleted by anybody if there are no instances of the contract on the chain, this would 
+break the `delegator` contract. To prevent this, the `delegator` contract can utilize the 
```

Comment:
I would add a simple ASCII diagram to illustrate these lock dependencies.

Reply:
How would the ASCII diagram look? It is a simple `1-*` dependency so maybe doesn't add much?

```
contractA ------------> 0x<code_hash>
contractB ------------|
```

Reply:
What if you have multiple nested dependencies?

Reply:
I don't see anything special about nested cross-contract deps or cycles. One contract can create locks and remove them, but they are always bound to the one creating contract.